digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/TrackCalculatorStrategy.scala

```diff
@@ -211,22 +212,58 @@ trait TrackCalculatorStrategy {
     pl.copy(calibrationPoints = CalibrationPointsUtils.toProjectLinkCalibrationPointsWithSourceInfo((sCP, eCP), source))
   }
 
-  protected def getUntilNearestAddress(seq: Seq[ProjectLink], endProjectLink: ProjectLink): (Seq[ProjectLink], Seq[ProjectLink]) = {
-    if (endProjectLink.discontinuity == MinorDiscontinuity || endProjectLink.discontinuity == Discontinuous) {
-      val continuousProjectLinks = seq.takeWhile(pl => pl.startAddrMValue < endProjectLink.endAddrMValue)
+  private def endPointOfProjectLink(pl: ProjectLink): Point = {
+    if (pl.sideCode == SideCode.TowardsDigitizing) {
+      pl.geometry.last
+    } else {
+      pl.geometry.head
+    }
+  }
+
+  private def lastSegmentDirection(pl: ProjectLink): Vector3d = {
+    if (pl.sideCode == SideCode.TowardsDigitizing) {
+      GeometryUtils.lastSegmentDirection(pl.geometry)
+    } else {
+      GeometryUtils.firstSegmentDirection(pl.geometry)
+    }
+  }
 
-      if (continuousProjectLinks.isEmpty)
+  /**
+    * Returns project links for the other track before and after the point where there is discontinuity on the track.
+    *
+    * @param trackA
+    * @param linkOnTrackB
+    * @return (Project links before the discontinuity point, project links after the discontinuity point)
+    */
+  protected def getUntilNearestAddress(trackA: Seq[ProjectLink], linkOnTrackB: ProjectLink): (Seq[ProjectLink], Seq[ProjectLink]) = {
+    if (linkOnTrackB.discontinuity == MinorDiscontinuity || linkOnTrackB.discontinuity == Discontinuous) {
+
+      // Sort links in order by the distance from the end of the link on track b.
+      val linkOnTrackBEndPoint = endPointOfProjectLink(linkOnTrackB)
```

Comment:
there is already defined methods for the first and last points for e.g.:
linkOnTrackB.getLastPoint
or
linkOnTrackB.getEndPoints()._2 (this one also checks if the link is reversed)

Reply:
Thanks, I'll change it!

Reply:
Fixed.